wgsl/index.bs

```diff
@@ -970,7 +972,14 @@ Restrictions on runtime-sized arrays:
     a store type in any other cases.
 * An expression must not evaluate to a runtime-sized array type.
 
-Issue: (dneto): Complete description of `Array<E,N>`
+<pre class='def'>
+array_type_decl
+  | attribute_list* ARRAY LESS_THAN type_decl (COMMA array_size_expr)? GREATER_THAN
+
+array_size_expr
```

Comment:
This constrains the expression by grammar.
I could have used literal_or_ident which is defined in the "Attributes" section, and then constrained with a validation rule.  I could go either way.